pyomo/common/config.py

```diff
@@ -2508,8 +2572,8 @@ class ConfigDict(ConfigBase, Mapping):
 
     content_filters = {None, 'all', 'userdata'}
 
-    __slots__ = ('_declared', '_implicit_declaration', '_implicit_domain')
-    _all_slots = set(__slots__ + ConfigBase.__slots__)
+    __slots__ = ('_implicit_domain', '_declared', '_implicit_declaration')
```

Comment:
Did the order matter?

Reply:
Yes.  `__getstate__` processes all `__slots__` except for `_implicit_domain` in a general manner (`_implicit_domain` is handled separately).  This processing works by skipping the first element in `__slots__`.  See the Note on lines 2607-2608.